adapter.js

```diff
@@ -106,42 +106,50 @@ export function adapter(svcName, aws) {
         .map(prop(name))
         .map(assoc("queue", name))
         .map(assoc("job", job))
-        .chain((msg) =>
-          getQueueUrl(svcName)
-            .chain((url) => sendMessage(url, msg))
-            .map(({ MessageId }) => ({
-              ok: true,
-              id: MessageId,
-              msg: assoc("status", "READY", msg),
-            }))
-        )
-        .chain((result) =>
-          putObject(svcName, `${name}/${result.id}`, result.msg).map(() =>
-            result
-          )
-        )
+        .chain(postJob(name))
         .toPromise(),
     // get jobs
     get: ({ name, status }) =>
       listObjects(svcName, name)
         .chain(includeDocs)
-        .map(filter(propEq('status', status)))
-        .map(jobs => ({ ok: true, jobs, status }))
-        .toPromise()
-    ,
+        .map(filter(propEq("status", status)))
+        .map((jobs) => ({ ok: true, jobs, status }))
+        .toPromise(),
     // retry job
-    retry: noop,
+    retry: ({ name, id }) =>
+      getObject(svcName, `${name}/${id}`)
+        .chain(({ job }) => postJob(name)(job))
```

Comment:
Should there be a check on the `status` being `READY` here?

Reply:
Retry would be for `ERROR`, I could add a check, to be safe, but the status READY should exist only in the time frame of < sec and the webhook target should be `idempotent` so multiple pushes of the same job is not a big deal, but I can add it, if it makes you feel better :)